cdap-app-fabric/src/main/java/io/cdap/cdap/internal/capability/CapabilityApplier.java

```diff
@@ -286,6 +312,135 @@ private void deployApp(SystemApplication application) throws Exception {
                  application.getArtifact(), configString, NOOP_PROGRAM_TERMINATOR, null, null);
   }
 
+  @VisibleForTesting
+  void autoInstallResources(String capability, List<AutoInstall> autoInstall) {
+    if (autoInstall == null || autoInstall.isEmpty()) {
+      LOG.debug("Capability {} do not have auto install resources associated with it", capability);
+      return;
+    }
+
+    for (AutoInstall a : autoInstall) {
```

Comment:
This method happen in a single thread. It is better to have the download and install happening concurrently.

Reply:
Fixed to execute concurrently